### PR TITLE
refac: project modernization

### PR DESCRIPTION
**Pull request by maicongodinho**
- Update project to Angular 13
- Restructured project based on new project generated with Angular CLI 13
- Exchanged TSLint for ESLint
- Exchanged Protractor for Cypress
- Update general outdated packages
- Minor layout fixes

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,35 +1,39 @@
...
-	},
-	{
-		path: 'link1',
-		component: Link1Component
-	}
+  // Lazy loading route
+  {
+    path: '',
+    loadChildren: () =>
+      import('./starter-home/starter-home.module').then(
```
Comment (zvonimirfras) [inline src/app/app-routing.module.ts:13]: This looks more readable as a oneliner

File: src/app/app.component.scss [hunk 2]
```diff
@@ -1 +1,5 @@
-@import "../assets/scss/scss-utils.scss";
+@import "~carbon-components/scss/globals/scss/spacing";
+
+.bx--content {
```
Comment (zvonimirfras) [inline src/app/app.component.scss:3]: Why is this needed?

File: src/app/app.component.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import { Component } from '@angular/core';
 
 @Component({
-	selector: 'app-root',
-	templateUrl: './app.component.html',
-	styleUrls: ['./app.component.scss']
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss'],
```
Comment (zvonimirfras) [inline src/app/app.component.ts:6]: Let's loose the trailing commas where not necessary. There used to be a lint rule for that. Not sure what happened to it.

File: src/app/app.module.ts [hunk 4]
```diff
@@ -1,46 +1,50 @@
...
 export class AppModule {
-	constructor(protected iconService: IconService) {
-		iconService.registerAll([
-			Notification20,
-			UserAvatar20,
-			AppSwitcher20
-		]);
-	}
+  constructor(protected iconService: IconService) {
+    iconService.registerAll([Notification16, UserAvatar16, Apps16]);
```
Comment (zvonimirfras) [inline src/app/app.module.ts:48]: 🙌

File: src/app/pages/link1/link1.component.html [hunk 5]
```diff
@@ -1 +1 @@
-<p>Link 1 page content goes here..</p>
+<p>link1 works!</p>
```
Comment (zvonimirfras) [inline src/app/pages/link1/link1.component.html:1]: Why not stay consistent with the content on other pages?

File: src/assets/scss/carbon-override.scss [hunk 6]
```diff
@@ -1,4 +0,0 @@
-// forces the navigation items to be displayed at all viewport sizes
```
Comment (zvonimirfras) [inline src/assets/scss/carbon-override.scss:1]: Keeping these files in scss could provide a useful structure for project organization, even if they aren't particularly useful in an empty project.

Is there something else that the starter is proposing instead?

File: src/index.html [hunk 7]
```diff
@@ -1,14 +1,13 @@
...
-</body>
+  <head>
+    <meta charset="utf-8" />
+    <title>Carbon Angular Starter</title>
+    <base href="/" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="icon" type="image/x-icon" href="favicon.ico" />
+  </head>
+  <body>
+    <app-root></app-root>
```
Comment (zvonimirfras) [inline src/index.html:11]: Why not
```suggestion
    <app-root>Loading...</app-root>
```
?

File: src/main.ts [hunk 8]
```diff
@@ -5,16 +5,8 @@ import { AppModule } from './app/app.module';
 import { environment } from './environments/environment';
 
 if (environment.production) {
-	enableProdMode();
+  enableProdMode();
 }
 
-// Uncomment the following lines to activate the service worker
```
Comment (zvonimirfras) [inline src/main.ts:11]: Why remove the service worker instruction?

File: src/typings.d.ts [hunk 9]
```diff
@@ -1,4 +1,4 @@
-/* SystemJS module definition */
-declare var module: {
-	id: string;
-};
+// icon modules goes here for strict applications
+declare module '@carbon/icons/es/notification/16';
```
Comment (zvonimirfras) [inline src/typings.d.ts:2]: There could be a catch-all module declaration for icons so they don't need to add one by one?

**Review by zvonimirfras — CHANGES_REQUESTED** [review]
Thank you for doing this, Maicon!

Here's a first pass, overall a good PR. I've got some suggestion and a few questions.

File: src/app/pages/link1/link1.component.html (hunk 5 above)
Comment (maicongodinho) [reply src/app/pages/link1/link1.component.html:1]: I will correct.

File: .gitignore [hunk 10]
```diff
@@ -1,41 +1,45 @@
...
-/e2e/*.js
-/e2e/*.map
-
-#System Files
+# System files
 .DS_Store
 Thumbs.db
+
+# e2e videos
+*.mp4
```
Comment (zvonimirfras) [inline .gitignore:45]:
```suggestion
*.mp4

```

**Review by Akshat55 — CHANGES_REQUESTED** [review]

File: src/app/header/header.component.ts [hunk 11]
```diff
@@ -3,9 +3,9 @@ import { Component, HostBinding } from '@angular/core';
 @Component({
 	selector: 'app-header',
 	templateUrl: './header.component.html',
-	styleUrls: ['./header.component.scss']
+	styleUrls: ['./header.component.scss'],
```
Comment (Akshat55) [inline src/app/header/header.component.ts:6]:
```suggestion
	styleUrls: ['./header.component.scss']
```

File: src/app/header/header.component.ts [hunk 12]
```diff
@@ -3,9 +3,9 @@ import { Component, HostBinding } from '@angular/core';
...
 })
 export class HeaderComponent {
-	// adds padding to the top of the document, so the content is below the header
 	@HostBinding('class.bx--header') headerClass = true;
+
```
Comment (Akshat55) [inline src/app/header/header.component.ts:10]:
```suggestion
	@HostBinding('class.bx--header') headerClass = true;
```

Comment (Akshat55) [reply src/app/header/header.component.ts:10]: Remove the empty line after the `@HostBinding`

File: src/app/starter-home/starter-home/starter-home.component.html [hunk 13]
```diff
@@ -26,47 +28,84 @@ <h1 class="title">Congratulations, <br>you made it!</h1>
...
+					</ibm-clickable-tile>
 				</div>
-				<div ibmCol [columnNumbers]="{'md': 2, 'sm': 12}">
-					<ibm-clickable-tile href="https://angular.carbondesignsystem.com/" target="_blank">Component Library</ibm-clickable-tile>
+				<div ibmCol [columnNumbers]="{ md: 2, sm: 12 }">
+					<ibm-clickable-tile
+						href="https://angular.carbondesignsystem.com/"
+						target="_blank"
+						>Component Library</ibm-clickable-tile
+					>
```
Comment (Akshat55) [inline src/app/starter-home/starter-home/starter-home.component.html:53]:
```suggestion
					<ibm-clickable-tile
						href="https://angular.carbondesignsystem.com/"
						target="_blank">
						Component Library
					</ibm-clickable-tile>
```

File: src/app/starter-home/starter-home/starter-home.component.html [hunk 14]
```diff
@@ -26,47 +28,84 @@ <h1 class="title">Congratulations, <br>you made it!</h1>
...
 				</div>
-				<div ibmCol [columnNumbers]="{'md': 2, 'sm': 12}">
-					<ibm-clickable-tile href="https://angular.carbondesignsystem.com/documentation/" target="_blank">Documentation</ibm-clickable-tile>
+				<div ibmCol [columnNumbers]="{ md: 2, sm: 12 }">
+					<ibm-clickable-tile
+						href="https://angular.carbondesignsystem.com/documentation/"
+						target="_blank"
+						>Documentation</ibm-clickable-tile
+					>
```
Comment (Akshat55) [inline src/app/starter-home/starter-home/starter-home.component.html:60]:
```suggestion
					<ibm-clickable-tile
						href="https://angular.carbondesignsystem.com/documentation/"
						target="_blank">
						Documentation
					</ibm-clickable-tile>
```

File: src/app/starter-home/starter-home/starter-home.component.html [hunk 15]
```diff
@@ -26,47 +28,84 @@ <h1 class="title">Congratulations, <br>you made it!</h1>
...
 				</div>
-				<div ibmCol [columnNumbers]="{'md': 2, 'sm': 12}">
-					<ibm-clickable-tile href="https://www.npmjs.com/package/carbon-components-angular" target="_blank">NPM</ibm-clickable-tile>
+				<div ibmCol [columnNumbers]="{ md: 2, sm: 12 }">
+					<ibm-clickable-tile
+						href="https://www.npmjs.com/package/carbon-components-angular"
+						target="_blank"
+						>NPM</ibm-clickable-tile
+					>
```
Comment (Akshat55) [inline src/app/starter-home/starter-home/starter-home.component.html:67]:
```suggestion
					<ibm-clickable-tile
						href="https://www.npmjs.com/package/carbon-components-angular"
						target="_blank">
						NPM
					</ibm-clickable-tile>
```

File: src/app/starter-home/starter-home/starter-home.component.html [hunk 16]
```diff
@@ -26,47 +28,84 @@ <h1 class="title">Congratulations, <br>you made it!</h1>
...
+			<br />
 			<div ibmRow>
-				<div ibmCol [columnNumbers]="{'md': 2, 'sm': 12}">
-					<ibm-clickable-tile href="https://github.com/IBM/carbon-charts" target="_blank">Repository</ibm-clickable-tile>
+				<div ibmCol [columnNumbers]="{ md: 2, sm: 12 }">
+					<ibm-clickable-tile
+						href="https://github.com/IBM/carbon-charts"
+						target="_blank"
+						>Repository</ibm-clickable-tile
+					>
```
Comment (Akshat55) [inline src/app/starter-home/starter-home/starter-home.component.html:87]:
```suggestion
					<ibm-clickable-tile
						href="https://github.com/IBM/carbon-charts"
						target="_blank">
						Repository
					</ibm-clickable-tile>
```

File: src/app/starter-home/starter-home/starter-home.component.html [hunk 17]
```diff
@@ -26,47 +28,84 @@ <h1 class="title">Congratulations, <br>you made it!</h1>
...
 				</div>
-				<div ibmCol [columnNumbers]="{'md': 2, 'sm': 12}">
-					<ibm-clickable-tile href="https://carbon-design-system.github.io/carbon-charts/" target="_blank">Demo</ibm-clickable-tile>
+				<div ibmCol [columnNumbers]="{ md: 2, sm: 12 }">
+					<ibm-clickable-tile
+						href="https://carbon-design-system.github.io/carbon-charts/"
+						target="_blank"
+						>Demo</ibm-clickable-tile
+					>
```
Comment (Akshat55) [inline src/app/starter-home/starter-home/starter-home.component.html:94]:
```suggestion
					<ibm-clickable-tile
						href="https://carbon-design-system.github.io/carbon-charts/"
						target="_blank">
						Demo
					</ibm-clickable-tile>
```

File: src/app/starter-home/starter-home/starter-home.component.html [hunk 18]
```diff
@@ -26,47 +28,84 @@ <h1 class="title">Congratulations, <br>you made it!</h1>
...
 				</div>
-				<div ibmCol [columnNumbers]="{'md': 2, 'sm': 12}">
-					<ibm-clickable-tile href="https://carbon-design-system.github.io/carbon-charts/documentation/" target="_blank">Documentation</ibm-clickable-tile>
+				<div ibmCol [columnNumbers]="{ md: 2, sm: 12 }">
+					<ibm-clickable-tile
+						href="https://carbon-design-system.github.io/carbon-charts/documentation/"
+						target="_blank"
+						>Documentation</ibm-clickable-tile
+					>
```
Comment (Akshat55) [inline src/app/starter-home/starter-home/starter-home.component.html:101]:
```suggestion
					<ibm-clickable-tile
						href="https://carbon-design-system.github.io/carbon-charts/documentation/"
						target="_blank">
						Documentation
					</ibm-clickable-tile>
```

File: src/app/starter-home/starter-home/starter-home.component.html [hunk 19]
```diff
@@ -26,47 +28,84 @@ <h1 class="title">Congratulations, <br>you made it!</h1>
...
 				</div>
-				<div ibmCol [columnNumbers]="{'md': 2, 'sm': 12}">
-					<ibm-clickable-tile href="https://www.npmjs.com/package/@carbon/charts" target="_blank">NPM</ibm-clickable-tile>
+				<div ibmCol [columnNumbers]="{ md: 2, sm: 12 }">
+					<ibm-clickable-tile
+						href="https://www.npmjs.com/package/@carbon/charts"
+						target="_blank"
+						>NPM</ibm-clickable-tile
+					>
```
Comment (Akshat55) [inline src/app/starter-home/starter-home/starter-home.component.html:108]:
```suggestion
					<ibm-clickable-tile
						href="https://www.npmjs.com/package/@carbon/charts"
						target="_blank">
						NPM
					</ibm-clickable-tile>
```

File: src/app/starter-home/starter-home/starter-home.component.spec.ts [hunk 20]
```diff
@@ -16,14 +16,14 @@ describe('HomeComponent', () => {
...
-			],
-			declarations: [ StarterHomeComponent ]
-		})
-		.compileComponents();
+			    GridModule, 
+			    ListModule, 
+			    TabsModule, 
+			    TilesModule
+		      ],
+
```
Comment (Akshat55) [inline src/app/starter-home/starter-home/starter-home.component.spec.ts:24]:
```suggestion
```

Remove blank line

**Review by Akshat55 — CHANGES_REQUESTED** [review]

File: src/app/app.component.spec.ts [hunk 21]
```diff
@@ -1,27 +1,38 @@
...
 import { HeaderComponent } from './header/header.component';
+import { CatalogComponent } from './pages/catalog/catalog.component';
+import { DocsComponent } from './pages/docs/docs.component';
+import { SupportComponent } from './pages/support/support.component';
 
 describe('AppComponent', () => {
-	beforeEach(waitForAsync(() => {
-		TestBed.configureTestingModule({
+	beforeEach(async () => {
+		await TestBed.configureTestingModule({
```
[the remark this answers is not in the document]
Comment (Akshat55) [reply src/app/app.component.spec.ts:16]: RESOVLED! I can't seem to resolve this for some reason - Seems like using async/await is now the new standard instead of using waitForAsync.

File: src/main.ts (hunk 8 above)
Comment (Akshat55) [reply src/main.ts:11]: RESOLVED

**Review by Akshat55 — APPROVED** [review]
